hydra-redis/redisgraph_demo1.py

```diff
@@ -0,0 +1,294 @@
+import redis
+from redisgraph import Graph, Node, Edge
+import urllib.request as ur
+import json
+import re
+from re import compile as regex
+from httplib2 import Http
+from hydrus.hydraspec import doc_maker
+
+
+redis_con = redis.Redis(host='localhost', port=6379)
+redis_graph = Graph("apidoc", redis_con)
+
+
+class _MemCache(dict):
+
+    def __nonzero__(self):
+        # even if empty, a _MemCache is True
+        return True
+
+    def set(self, key, value):
+        self[key] = value
+
+    def delete(self, key):
+        if key in self:
+            del self[key]
+
+
+DEFAULT_HTTP_CLIENT = Http(_MemCache())
+http = DEFAULT_HTTP_CLIENT
+APIDOC_RE = regex(
+    r'^<([^>]*)>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"$')
+
+
+def final_file(url):
+    """ open the given url and read and load the Json data.
+    """
+    response = ur.urlopen(url)
+    return json.loads(response.read().decode('utf-8'))
+
+
+def objects_property(objects_node, new_list, url_node):
+    """makes nodes for those properties which have properties in itself.
+    """
+    print("for the properties which have class and should have node")
+    print(new_list)
+    for obj in new_list:
+        obj_method = []
+        obj_prop = []
+        properties = []
+        obj = obj.replace("vocab:", "")
+        objects_properties = {}
+        for obj1 in api_doc.parsed_classes[obj]["class"].supportedOperation:
```

Comment:
You are using a global variable without declaring it as global, you should pass the variable as a parameter instead.